### PR TITLE
INC-485: return default IEP level for a prison if there is no current IEP for a list of booking IDs

### PR DESCRIPTION
**Pull request by neilmendum**
https://dsdmoj.atlassian.net/browse/INC-485

### CODE REVIEW

File: src/test/java/uk/gov/justice/hmpps/prison/service/BookingServiceTest.java [hunk 1]
```diff
@@ -367,16 +367,37 @@ public void getBookingIEPSummary_offenderHasNoIepSummary() {
     public void getBookingIEPSummary_multipleBooking_globalSearchUser() {
         when(authenticationFacade.isOverrideRole(anyString(), anyString(), anyString(), anyString(), anyString())).thenReturn(true);
         when(bookingRepository.getBookingIEPDetailsByBookingIds(anyList())).thenReturn(Map.of(-5L, List.of(PrivilegeDetail.builder().iepDate(LocalDate.now()).build())));
+        when(offenderBookingRepository.findById(anyLong())).thenReturn(Optional.of(OffenderBooking.builder().location(AgencyLocation.builder().id("MDI").build()).build()));
```
Comment (neilmendum) [inline src/test/java/uk/gov/justice/hmpps/prison/service/BookingServiceTest.java:370]: Had to update existing tests because I introduced `offenderBookingRepository.findById` and without this mock they were throwing `EntityNotFoundException`

File: src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java [hunk 2]
```diff
@@ -342,16 +326,32 @@ public Map<Long, PrivilegeSummary> getBookingIEPSummary(final List<Long> booking
...
                 .filter(bookingId -> !mapOfEip.containsKey(bookingId))
-                .collect(toList())
-                .forEach(bookingId -> mapOfEip.put(bookingId, PrivilegeSummary.builder()
-                        .bookingId(bookingId)
-                        .iepLevel(defaultIepLevel)
-                        .iepDetails(Collections.emptyList())
-                        .build()));
+                .toList()
+                .forEach(bookingId -> {
+                    final var offenderBooking = offenderBookingRepository.findById(bookingId).orElseThrow(EntityNotFoundException.withId(bookingId));
```
Comment (neilmendum) [inline src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java:331]: Not ideal doing another lookup but it's needed so we can get a prison's default IEP. This is an edge case so not expecting it to called frequently

File: src/test/java/uk/gov/justice/hmpps/prison/service/BookingServiceTest.java (hunk 1 above)
Comment (Mjwillis) [reply src/test/java/uk/gov/justice/hmpps/prison/service/BookingServiceTest.java:370]: Does this definitely only get called if no IEP found?

Comment (Mjwillis) [reply src/test/java/uk/gov/justice/hmpps/prison/service/BookingServiceTest.java:370]: Ignore I can see that it does

**Review by Mjwillis — APPROVED** [review]
LGTM
